src/msgraph_core/authentication/azure_identity_authentication_provider.py

```diff
@@ -0,0 +1,36 @@
+from typing import TYPE_CHECKING, List, Union, Optional, Dict
+
+from kiota_authentication_azure.azure_identity_authentication_provider import \
+    AzureIdentityAuthenticationProvider as KiotaAzureIdentityAuthenticationProvider
+from msgraph_core._constants import MS_DEFAULT_SCOPE
+from msgraph_core._enums import NationalClouds
+
+if TYPE_CHECKING:
+    from azure.core.credentials import TokenCredential
+    from azure.core.credentials_async import AsyncTokenCredential
+
+
+class AzureIdentityAuthenticationProvider(KiotaAzureIdentityAuthenticationProvider):
+
+    def __init__(
+        self,
+        credentials: Union["TokenCredential", "AsyncTokenCredential"],
+        options: Optional[Dict] = None,
+        scopes: Optional[List[str]] = None,
+        allowed_hosts: Optional[List[str]] = None
+    ) -> None:
+        """[summary]
+
+        Args:
+            credentials (Union["TokenCredential", "AsyncTokenCredential"]): The
+                tokenCredential implementation to use for authentication.
+            options (Optional[dict]): The options to use for authentication.
+            scopes (Optional[List[str]]): The scopes to use for authentication.
+                Defaults to 'https://graph.microsoft.com/.default'.
+            allowed_hosts (Optional[List[str]]): The allowed hosts to use for
+                authentication. Defaults to Microsoft National Clouds.
+        """
+        self.scopes: List[str] = [MS_DEFAULT_SCOPE] if scopes is None else scopes
```

Comment:
I think scopes should default to an empty list instead of `https://graph.microsoft.com/.default`/None so that by default the [Kiota Access Token provider](https://github.com/microsoft/kiota-authentication-azure-python/blob/main/kiota_authentication_azure/azure_identity_access_token_provider.py#L52) builds the appropriate default scope. This handles setting default scopes when using other [Graph national cloud endpoints](https://learn.microsoft.com/en-us/graph/deployments)